src/software/geom/ray.h

```diff
@@ -49,16 +49,16 @@ class Ray final
 
     inline Segment toSegment() const
     {
-        return Segment(start, direction);
+        return Segment(start, Point(direction));
     }
 
     inline Vector toVector() const
     {
-        return direction - start;
+        return Point(direction) - start;
```

Comment:
Shouldn't this just be: 
```suggestion
        return direction;
```

Reply:
fixed in latest update